### PR TITLE
New resource: aws_cloudwatch_composite_alarm

### LINKED ISSUE #12272: CloudWatch composite alarms
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon CloudWatch now allows you to combine multiple alarms.

### New or Affected Resource(s)

* `aws_cloudwatch_composite_alarm`

### Potential Terraform Configuration

```hcl
resource "aws_cloudwatch_composite_alarm" "alarm" {
  alarm_name = "example"
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/03/amazon-cloudwatch-now-allows-you-to-combine-multiple-alarms/).
[User guide](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Create_Composite_Alarm.html).

### PR DESCRIPTION
**Pull request by roberth-k**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12272

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
FEATURES:
- **New Resource:** `aws_cloudwatch_composite_alarm`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAwsCloudWatchCompositeAlarm_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAwsCloudWatchCompositeAlarm_ -timeout 120m
--- PASS: TestAccAwsCloudWatchCompositeAlarm_disappears (19.78s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_basic (22.99s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_update (36.72s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       38.508s

```

### CODE REVIEW

**Comment by jvisker** [comment]
Any updates on this?

**Comment by jayy-mize** [comment]
any ETA on this feature being released?

**Comment by mqchau** [comment]
@breathingdust Could you please take a look at this?

**Comment by breathingdust** [comment]
Hi all 👋 . Just letting you know this is on our list for our engineers to review and will feature in an upcoming release.

**Review by anGie44 — CHANGES_REQUESTED** [review]
Hi @roberth-k, thank you for this PR ! Off to a really great start 👍 Please find my comments here for review. Feel free to reach out with any questions/concerns! 

Output of acceptance tests:
```
--- PASS: TestAccAwsCloudWatchCompositeAlarm_disappears (13.90s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_basic (37.54s)
```

**Comment by anGie44** [comment]
Hi @roberth-k 👋 checking in to see if you'll be able to address the changes requested here. Not a problem if you can't, we're happy to continue the work here and get it into an upcoming release, just let us know :)

**Comment by roberth-k** [comment]
Hi @anGie44! I'll get it sorted some time this week.

**Comment by anGie44** [comment]
sounds great, thanks @roberth-k 👍

File: aws/resource_aws_cloudwatch_composite_alarm.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	d *schema.ResourceData,
+	meta interface{},
+) diag.Diagnostics {
+	conn := meta.(*AWSClient).cloudwatchconn
+	name := d.Get("alarm_name").(string)
+
+	input := expandAwsCloudWatchPutCompositeAlarmInput(d)
+	_, err := conn.PutCompositeAlarmWithContext(ctx, &input)
+	if err != nil {
+		return diag.Errorf("create composite alarm: %s", err)
```
[the remark this answers is not in the document]
Comment (roberth-k) [reply aws/resource_aws_cloudwatch_composite_alarm.go]: Note in all cases I've retained the `%s` formatting directive, as acceptance testing with `%w` yields errors such as `aws/resource_aws_cloudwatch_composite_alarm.go:125:10: Errorf call has error-wrapping directive %w`.

**Comment by anGie44** [comment]
Hi @roberth-k 👋 Thank you again for this PR -- looked great overall! I've added a couple more acceptance tests for coverage to get this into our upcoming release. 

Output of acceptance tests:
```
--- PASS: TestAccAwsCloudWatchCompositeAlarm_disappears (183.81s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_basic (231.57s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_description (306.49s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_updateAlarmRule (325.89s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_allActions (326.25s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_actionsEnabled (326.34s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_insufficientDataActions (375.92s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_alarmActions (376.01s)
--- PASS: TestAccAwsCloudWatchCompositeAlarm_okActions (377.01s)
```

**Review by anGie44 — APPROVED** [review]

**Pull request merged**